operators/src/source/csv.rs

```diff
@@ -286,26 +284,37 @@ impl Stream for CsvSourceStream {
     type Item = Result<MultiPointCollection>;
 
     fn poll_next(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
-        let mut state = self.state.lock().unwrap(); // TODO: handle lock poisoning
-        if state.poll_result.is_some() {
-            let x = state.poll_result.take().unwrap();
+        // TODO: handle lock on multiple occasions
+
+        if self.thread_is_computing.load(Ordering::Relaxed) {
```

Comment:
why is this extra indicator needed? You could just check poll result being `Some` like before.

Reply:
No, because if you call `poll_next` for the first time, it is `None` and you have no clue that you should start the thread.
On the other hand, if you already started the thread, but did not wait for the notification to arrive, you want to get pending again and not start the thread a second time. I introduced this, since moving to `spawn_blocking` lead to test errors based on race conditions.

Reply:
wouldn't it make more sense to have a bool variable like `init` to check if it's the first run instead of updating this variable all the time?

Reply:
Well, then you have the problem a second and a third time.

* first poll, set this variable to true and start thread
* second (intermediate) poll, the result is None, so return pending
* thread is finished and calls waker
* fourth poll, you return Some(thing), leaving None in-place
* fifth poll, now you would like to start a new work thread, but you see that there is None in the variable, so you think there is already a thread.
* …